Stratis.Bitcoin/Utilities/CustomThreadPoolScheduler.cs

```diff
@@ -8,131 +8,196 @@
+        /// from the code.
+        /// </para>
+        /// </remarks>
+        private AutoResetEvent finished = new AutoResetEvent(false);
+
+        /// <summary>Number of scheduler threads that are currently not busy and can be used for task execution.</summary>
+        private int availableThreads;
+        /// <summary>Number of scheduler threads that are currently not busy and can be used for task execution.</summary>
+        public int AvailableThreads
+        {
+            get
+            {
+                return this.availableThreads;
+            }
+        }
+
+        /// <summary>Blocking queue of tasks to be executed by this scheduler.</summary>
+        private BlockingCollection<Task> tasks;
+
+        /// <summary>
+        /// Initializes a new instance of the object with a specific number of threads that 
+        /// the scheduler can use and a maximum limit on the capacity of the task queue.
+        /// </summary>
+        /// <param name="threadCount">Number of threads that the scheduler can use to execute the tasks.</param>
+        /// <param name="maxQueued">Maximum number of queued tasks.</param>
+        /// <param name="name">Name of the thread that is responsible for executing the tasks by this scheduler. See <see cref="Thread.Name"/>.</param>
+        public CustomThreadPoolTaskScheduler(int threadCount, int maxQueued, string name = null)
+        {
+            this.threadCount = threadCount;
+            this.tasks = new BlockingCollection<Task>(new ConcurrentQueue<Task>(), maxQueued);
+            this.availableThreads = threadCount;
+            for (int i = 0; i < threadCount; i++)
+            {
+                new Thread(Do)
+                {
+                    IsBackground = true,
+                    Name = name
+                }.Start();
+            }
+        }
+
+        /// <inheritdoc />
+        public override int MaximumConcurrencyLevel
+        {
+            get
+            {
+                return this.threadCount;
+            }
+        }
+
+        /// <summary>Number of unfinished tasks that are waiting to be executed.</summary>
+        public int QueuedCount
+        {
+            get
+            {
+                return this.tasks.Count;
+            }
+        }
+
+        /// <summary>Number of tasks that remains to be finished. This is number of queued tasks waiting for execution plus number of currently executing tasks.</summary>
+        public int RemainingTasks
+        {
+            get
+            {
+                return (this.threadCount - this.AvailableThreads) + this.QueuedCount;
+            }
+        }
+
+        /// <summary>
+        /// Thread procedure that consumes the tasks added to the queue and executes them.
+        /// <para>There are <see cref="threadCount"/> number of threads executing this method at the same time.</para>
+        /// </summary>
+        void Do()
```

Comment:
It is, but tend to only rename stuff while xmldocing when it is against code style.
Do you want me to change it here for this PR?